hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/ipc/TestNettyRPC.java

```diff
@@ -0,0 +1,1742 @@
+/**
```

Comment:
Is this largely the same as TestRPC? Due to the length of the test file, this is impossible to review. Any chance to merge it with TestRPC?

Reply:
You are correct, this is TestRPC with netty turned on. I noticed that parameterizing the unit tests and running both nettya and non-netty was causing a lot of test failures. These failures just went away when we separated the netty and non-netty tests.

I thought as a first step I will just duplicate the test in a separate file and check whether things pass. This seems to work. A lot of the failures have gone away.

In my next PR, I will eliminate the redundancy.

Sorry for the inconvenience. But please rest assure that the test code will pretty much remain the same.